Bottom: 82bb08138c6d436da67ee5ec6d655691df613629
Top:    58d7016c3c884d3782c3596d5ece58c2a54df64f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 19:36:09 +0000

Refresh of tmp

---

diff --git a/src/polygon1.c b/src/polygon1.c
index a0a0abb..08d74b1 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -915,48 +915,6 @@ cntrbox_inside (PLINE * c1, PLINE * c2)
 /*****************************************************************/
 /* Routines for making labels */
 
-/* cntr_in_M_PLINE
-returns poly is inside outfst ? TRUE : FALSE */
-static int
-cntr_in_M_PLINE (PLINE * poly, PLINE * outfst, BOOLp test)
-{
-  PLINE *curc;
-  PLINE *outer = outfst;
-  heap_t *heap;
-
-  assert (poly != NULL);
-  assert (outer != NULL);
-
-  heap = heap_create ();
-//  do {
-    if (cntrbox_inside (poly, outer))
-      heap_insert (heap, outer->area, (void *) outer);
-//  /* if checking touching, use only the first polygon */
-//  } while (!test && (outer = outer->f) != outfst);
-
-  /* we need only check the smallest poly container
-   * but we must loop in case the box containter is not
-   * the poly container */
-  while (1) {
-    if (heap_is_empty (heap))
-      break;
-    outer = (PLINE *) heap_remove_smallest (heap);
-    if (poly_ContourInContour (outer, poly)) {
-      for (curc = outer->next; curc != NULL; curc = curc->next)
-        if (poly_ContourInContour (curc, poly)) {
-          /* it's inside a hole in the smallest polygon
-           * no need to check the other polygons */
-          heap_destroy (&heap);
-          return FALSE;
-        }
-      heap_destroy (&heap);
-      return TRUE;
-    }
-  }
-  heap_destroy (&heap);
-  return FALSE;
-}				/* cntr_in_M_PLINE */
-
 /* cntr_in_M_POLYAREA
 returns poly is inside outfst ? TRUE : FALSE */
 static int
@@ -1088,63 +1046,6 @@ label_contour (PLINE * a)
 }				/* label_contour */
 
 static BOOLp
-cntr_label_PLINE (PLINE * poly, PLINE * pl, BOOLp test)
-{
-  assert (ppl != NULL);
-  if (poly->Flags.status == ISECTED) {
-    printf ("cntr_label_PLINE: Labelling intersected contour\n");
-    label_contour (poly);	/* should never get here when BOOLp is true */
-
-  } else {
- 
-    fprintf (stderr, "*********** SOMETHING ODD GOING ON HERE?\n");
-//    return FALSE;
-
-    if (cntr_in_M_PLINE (poly, pl, test)) {
-
-      if (test)
-        return TRUE;
-      poly->Flags.status = INSIDE;
-
-    } else {
-
-      if (test)
-        return False;
-      if (poly->Flags.status == UNKNWN) {
-        printf ("cntr_label_PLINE: Changing UNKNWN to OUTSIDE\n");
-        poly->Flags.status = OUTSIDE;
-      }
-
-    }
-  }
-  return FALSE;
-}				/* cntr_label_PLINE */
-
-static BOOLp
-cntr_label_POLYAREA_non_isected (PLINE * poly, POLYAREA * ppl, BOOLp test)
-{
-  assert (ppl != NULL && ppl->contours != NULL);
-  if (poly->Flags.status == ISECTED)
-    {
-      printf ("cntr_label_POLYAREA_non_isected: Skipping labelling intersected contour\n");
-      //label_contour (poly);	/* should never get here when BOOLp is true */
-    }
-  else if (cntr_in_M_POLYAREA (poly, ppl, test))
-    {
-      if (test)
-	return TRUE;
-      poly->Flags.status = INSIDE;
-    }
-  else
-    {
-      if (test)
-	return False;
-      poly->Flags.status = OUTSIDE;
-    }
-  return FALSE;
-}				/* cntr_label_POLYAREA */
-
-static BOOLp
 cntr_label_POLYAREA (PLINE * poly, POLYAREA * ppl, BOOLp test)
 {
   assert (ppl != NULL && ppl->contours != NULL);
@@ -1182,47 +1083,6 @@ M_POLYAREA_label_separated (PLINE * afst, POLYAREA * b, BOOLp touch)
 }
 
 static BOOLp
-M_POLYAREA_label_isected (POLYAREA * afst, PLINE * b, BOOLp touch)
-{
-  POLYAREA *a = afst;
-  PLINE *curc;
-
-  if (b == NULL) {
-    printf ("M_POLYAREA_label_isected: No PLINE to test against\n");
-    return FALSE;
-  }
-
-  assert (a != NULL);
-  do {
-    for (curc = a->contours; curc != NULL; curc = curc->next)
-      if (cntr_label_PLINE (curc, b, touch) && touch)
-        return TRUE;
-  } while (!touch && (a = a->f) != afst);
-  return FALSE;
-}
-
-
-static BOOLp
-M_POLYAREA_label_non_isected (POLYAREA * afst, POLYAREA * b, BOOLp touch)
-{
-  POLYAREA *a = afst;
-  PLINE *curc;
-
-  assert (a != NULL);
-  do
-    {
-      for (curc = a->contours; curc != NULL; curc = curc->next)
-	if (cntr_label_POLYAREA_non_isected (curc, b, touch))
-	  {
-	    if (touch)
-	      return TRUE;
-	  }
-    }
-  while (!touch && (a = a->f) != afst);
-  return FALSE;
-}
-
-static BOOLp
 M_POLYAREA_label (POLYAREA * afst, POLYAREA * b, BOOLp touch)
 {
   POLYAREA *a = afst;
@@ -2158,7 +2018,6 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
       M_POLYAREA_label (b, a, FALSE);
 #if 0
       M_POLYAREA_label_separated (a_isected, b, FALSE);
-      M_POLYAREA_label_isected (b, a_isected, FALSE);
 #endif
 
 /* New faster method */
